XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/BaseHand.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using XRTK.Definitions.Devices;
+using XRTK.Definitions.Utilities;
+using XRTK.Interfaces.InputSystem;
+using XRTK.Interfaces.Providers.Controllers;
+
+namespace XRTK.Providers.Controllers.Hands
+{
+    public abstract class BaseHand : BaseController, IMixedRealityHand
+    {
+        // Hand ray
+        protected HandRay HandRay { get; } = new HandRay();
+
+        public override bool IsInPointingPose
+        {
+            get
+            {
+                return HandRay.ShouldShowRay;
+            }
+        }
+
+        // Velocity internal states
+        private float deltaTimeStart;
+        private Vector3 lastPosition;
+        private Vector3 lastPalmNormal;
+        private readonly int velocityUpdateInterval = 9;
+        private int frameOn = 0;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="trackingState"></param>
+        /// <param name="controllerHandedness"></param>
+        /// <param name="inputSource"></param>
+        /// <param name="interactions"></param>
+        public BaseHand(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+                : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+        }
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultLeftHandedInteractions => DefaultInteractions;
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultRightHandedInteractions => DefaultInteractions;
+
+        public override void SetupDefaultInteractions(Handedness controllerHandedness)
+        {
+            AssignControllerMappings(DefaultInteractions);
+        }
+
+        #region Protected InputSource Helpers
+
+        #region Gesture Definitions
+
+        protected void UpdateVelocity()
+        {
+            if (frameOn == 0)
+            {
+                deltaTimeStart = Time.unscaledTime;
+                lastPosition = GetJointPosition(TrackedHandJoint.Palm);
+                lastPalmNormal = GetPalmNormal();
+                //lastPalmNormal = new Vector3(1, 0, 0);
+            }
+            else if (frameOn == velocityUpdateInterval)
+            {
+                //update linear velocity
+                float deltaTime = Time.unscaledTime - deltaTimeStart;
+                Vector3 newVelocity = (GetJointPosition(TrackedHandJoint.Palm) - lastPosition) / deltaTime;
+                Velocity = (Velocity * 0.8f) + (newVelocity * 0.2f);
```

Comment:
These constants should be specified as such.

Reply:
Thanks Eric!